### DISCUSSION TITLE
HDR features

### DISCUSSION THREAD

**Discussion by Alanko5** [kingslay/KSPlayer#271] [Ideas]
HDR is beautifully described in this article:
http://www.dev-qiyun.club/2022/02/11/如何渲染HDR图像或视频？/

I was interested in the upscaling from SDR to HDR:
-x265-params "colorprim=bt2020:transfer=smpte2084:colormatrix=bt2020nc:master-display=""G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,1)"":max-cll=""1000,200"":min-luma=0.005:max-luma=1000" 

Some films are in SDR and will never be like HDR.
If the user has an HDR TV, it would be nice for him to be able to turn it on.

And the other way around:
-vf zscale=t=linear:npl=100,format=gbrpf32le,zscale=p=bt709,tonemap=tonemap=hable:desat=0,zscale=t=bt709:m=bt709:r=tv,format=yuv420p -pix_fmt yuv420p -color_range tv -colorspace bt709 -color_trc bt709 -color_primaries bt709

Some users do not have HDR TV.
For them, it would be interesting to have HDR content played as SDR.
Because HDR content is dark enough for SDR TV.

What do you think?
Is such remapping possible during streaming?

**Comment by kingslay** [comment]
> HDR is beautifully described in this article: [http://www.dev-qiyun.club/2022/02/11/如何渲染HDR图像或视频？/](http://www.dev-qiyun.club/2022/02/11/%E5%A6%82%E4%BD%95%E6%B8%B2%E6%9F%93HDR%E5%9B%BE%E5%83%8F%E6%88%96%E8%A7%86%E9%A2%91%EF%BC%9F/)
> 
> I was interested in the upscaling from SDR to HDR: -x265-params "colorprim=bt2020:transfer=smpte2084:colormatrix=bt2020nc:master-display=""G(13250,34500)B(7500,3000)R(34000,16000)WP(15635,16450)L(10000000,1)"":max-cll=""1000,200"":min-luma=0.005:max-luma=1000"
> 
> Some films are in SDR and will never be like HDR. If the user has an HDR TV, it would be nice for him to be able to turn it on.
> 
> And the other way around: -vf zscale=t=linear:npl=100,format=gbrpf32le,zscale=p=bt709,tonemap=tonemap=hable:desat=0,zscale=t=bt709:m=bt709:r=tv,format=yuv420p -pix_fmt yuv420p -color_range tv -colorspace bt709 -color_trc bt709 -color_primaries bt709
> 
> Some users do not have HDR TV. For them, it would be interesting to have HDR content played as SDR. Because HDR content is dark enough for SDR TV.
> 
> What do you think? Is such remapping possible during streaming?

it is a good idea. I will try to use videoFilters to achieve it. 题外话，你懂得中文吗？因为我看你分享的链接是中文的

**Reply by Alanko5** [reply]
I don't know, but google translator knows. 😉

**Comment by Alanko5** [comment]
https://developer.apple.com/av-foundation/Incorporating-HDR-video-with-Dolby-Vision-into-your-apps.pdf

Apple Doc:
**How to Convert HDR Video to SDR Video**
Down-converting 10-bit HDR to 8-bit SDR involves two independent transformations: HDR to SDR color space conversion and 10-bit to 8-bit bit-depth conversion. You need to ensure that you perform the color conversion before or, optimally, at the same time as the 10-bit to 8-bit pixel format conversion. If 10-bit to 8-bit pixel format conversion occurs before the color conversion, then undesirable banding artifacts will be present in the output video.
For the color space conversion, the only thing you need to do is explicitly specify an SDR color space and an 8-bit codec and profile level. A general rule of thumb is to set the color properties to BT.709. Specifically, if you include video composition in your playback or editing, you would set the following SDR color properties in your AVVideoComposition.
videoComposition.colorPrimaries = AVVideoColorPrimaries_ITU_R_709_2
videoComposition.colorTransferFunction =
AVVideoTransferFunction_ITU_R_709_2 videoComposition.colorYCbCrMatrix
= AVVideoYCbCrMatrix_ITU_R_709_2
